src/pages/layout.jsx

```diff
@@ -46,39 +50,52 @@ export default function Layout({ children }) {
   };
 
   useEffect(() => {
-    let dayobsStr = getDayobsStr(dayobs);
+    setExposuresLoading(true);
+    setAlmanacLoading(true);
+    setNarrativeLoading(true);
 
+    let dayobsStr = getDayobsStr(dayobs);
     if (!dayobsStr) {
       console.error("No Date Selected!");
+      setExposuresLoading(false);
+      setAlmanacLoading(false);
+      setNarrativeLoading(false);
       return;
     }
-
     let dateFromDayobs = getDatetimeFromDayobsStr(dayobsStr);
-
     let startDate = dateFromDayobs.minus({ days: noOfNights - 1 });
     let startDayobs = startDate.toFormat("yyyyLLdd");
-
     let endDate = dateFromDayobs.plus({ days: 1 });
     let endDayobs = endDate.toFormat("yyyyLLdd");
 
-    // Fetch exposures, almanac, and narrative log
-    fetchExposures(startDayobs, endDayobs, instrument).then(
-      ([exposuresNo, exposureTime]) => {
+    fetchExposures(startDayobs, endDayobs, instrument)
+      .then(([exposuresNo, exposureTime]) => {
         setExposureCount(exposuresNo);
         setSumExpTime(exposureTime);
-      },
-    );
-
-    fetchAlmanac(startDayobs, endDayobs).then((hours) => {
-      setNightHours(hours);
-    });
-
-    fetchNarrativeLog(startDayobs, endDayobs, instrument).then(
-      ([weather, fault]) => {
+        setExposuresLoading(false);
+      })
+      .catch(() => {
+        setExposuresLoading(false);
+      });
+
+    fetchAlmanac(startDayobs, endDayobs)
+      .then((hours) => {
+        setNightHours(hours);
+        setAlmanacLoading(false);
+      })
+      .catch(() => {
+        setAlmanacLoading(false);
+      });
```

Comment:
I think, instead of calling `setAlmanacLoading(false);` twice (in the `then` and `catch` calls), you could just have it one time with a `finally` call:
```suggestion
    fetchAlmanac(startDayobs, endDayobs)
      .then((hours) => {
        setNightHours(hours);
      })
      .finally(() => {
        setAlmanacLoading(false);
      });
```
This can be applied to `fetchExposures`, `fetchAlmanac` and `fetchNarrativeLog`.